app/src/main/java/net/sourceforge/opencamera/sensorlogging/VideoFrameInfo.java

```diff
@@ -156,15 +178,29 @@ public void submitProcessFrame(long timestamp, byte[] imageData, int width, int
     }
 
     private void writeFrameTimestamp(long timestamp) {
-        try {
-            mFrameBufferedWriter
-                    .append(Long.toString(timestamp))
-                    .append("\n");
-        } catch (IOException e) {
-            mAppInterface.onFrameInfoRecordingFailed();
-            Log.e(TAG, "Failed to write frame info, timestamp: " + timestamp);
-            e.printStackTrace();
-            this.close();
+        if (mUnsyncedFrameBufferedWriter != null) {
+            try {
+                mUnsyncedFrameBufferedWriter
+                        .append(Long.toString(timestamp))
+                        .append("\n");
+            } catch (IOException e) {
+                mAppInterface.onFrameInfoRecordingFailed();
+                Log.e(TAG, "Failed to write unsynced frame info, timestamp: " + timestamp);
+                e.printStackTrace();
+                this.close();
+            }
+        }
+        if (mSyncedFrameBufferedWriter != null) {
```

Comment:
Can we get rid of duplicating this by using some variables for different suffix/time, reusing the rest of the code?

Reply:
I'm not sure if I did what you meant in [this commit](https://github.com/MobileRoboticsSkoltech/OpenCamera-Sensors/pull/46/commits/b1f7ae9e5acc1d9d69b73f561aef97770331d3d6).